gnd/src/main/java/com/google/android/gnd/ui/home/mapcontainer/MapContainerViewModel.java

```diff
@@ -330,10 +388,38 @@ private MapGeoJson toMapGeoJson(GeoJsonFeature feature) {
   }
 
   public LiveData<CameraPosition> getCameraPosition() {
+    checkPointNearVertex(cameraPosition.getValue());
     Timber.d("Current position is %s", cameraPosition.getValue().toString());
     return cameraPosition;
   }
 
+  private void checkPointNearVertex(CameraPosition position) {
+    if (vertices.isEmpty() || vertices.size() < 3) {
+      return;
+    }
+    //  Experimentation with isPointNearFirstVertex() if does not work
+    //  replace vertices.contains(point)
+    if (vertices.get(0) == position.getTarget() || isPointNearFirstVertex(position.getTarget())) {
+      updatePolygonDrawing(PolygonDrawing.COMPLETED);
+      vertices.add(vertices.get(0));
+
+      updateDrawnPolygonFeature(ImmutableList.copyOf(vertices));
+    } else {
+      updatePolygonDrawing(PolygonDrawing.DEFAULT);
+    }
+  }
+
+  private boolean isPointNearFirstVertex(Point point) {
+    float[] distance = new float[1];
+    Location.distanceBetween(
+        point.getLatitude(),
+        point.getLongitude(),
+        vertices.get(0).getLatitude(),
+        vertices.get(0).getLongitude(),
+        distance);
+    return distance[0] < 10;
```

Comment:
Move distance threshold to a constant and document.

Reply:
moved to constant